server/src/entities/sso_config.entity.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  Entity,
+  Column,
+  PrimaryGeneratedColumn,
+  CreateDateColumn,
+  UpdateDateColumn,
+  ManyToOne,
+  JoinColumn,
+} from 'typeorm';
+import { Organization } from './organization.entity';
+
+interface Google {
+  clientId: string;
+}
+interface Git {
+  clientId: string;
+  clientSecret: string;
+}
+interface Okta {
+  clientId: string;
+  clientSecret: string;
+  domain: string;
+  redirectUri: string;
+  authServer?: string;
+}
+
+interface Form {
+  enableSignUp?: boolean;
+}
+
+@Entity({ name: 'sso_configs' })
+export class SSOConfigs {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column({ name: 'organization_id' })
+  organizationId: string;
+
+  @Column({ name: 'sso' })
+  sso: 'google' | 'git' | 'okta' | 'form';
+
+  @Column({ type: 'json' })
+  configs: Google | Git | Okta | Form | any;
```

Comment:
Nice! We should also be encrypting the client creds as well.

Reply:
we are encrypting any keys containing `secret`